Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolverCocoa.cpp

```diff
@@ -46,18 +47,12 @@ static void resolvedName(CFHostRef hostRef, CFHostInfoType typeInfo, const CFStr
     }
 
     Boolean result;
-    CFArrayRef resolvedAddresses = (CFArrayRef)CFHostGetAddressing(hostRef, &result);
+    CFArrayRef resolvedAddresses = CFHostGetAddressing(hostRef, &result);
     ASSERT_UNUSED(result, result);
 
-    size_t count = CFArrayGetCount(resolvedAddresses);
-    Vector<WebCore::IPAddress> addresses;
-    addresses.reserveInitialCapacity(count);
-
-    for (size_t index = 0; index < count; ++index) {
-        CFDataRef data = (CFDataRef)CFArrayGetValueAtIndex(resolvedAddresses, index);
-        if (auto address = WebCore::IPAddress::fromSockAddrIn6(*reinterpret_cast<const struct sockaddr_in6*>(CFDataGetBytePtr(data))))
-            addresses.uncheckedAppend(*address);
-    }
+    auto addresses = makeVector<CFDataRef>(resolvedAddresses, [] (CFDataRef data) -> std::optional<WebCore::IPAddress> {
+        return WebCore::IPAddress::fromSockAddrIn6(*reinterpret_cast<const struct sockaddr_in6*>(CFDataGetBytePtr(data)));
+    });
```

Comment:
I feel like the `makeVector` template for this function should be able to figure out both the `Vector<>` element type and the `CFType` using the functor, but I haven't figured out how to write that in C++ template code.

It's a little non-intuitive (and weird) to write `makeVector<CFDataRef>` here when the output is a `Vector<WebCore::IPAddress>`.  (I could change it back to makeVector<WebCore::IPAddress, CFDataRef>`, but the functor should have all the type info we need!)

Reply:
I think I figured this out.  Had to think differently about how to declare the arguments to `MapFunctionType` in the template.

Reply:
Or not.  Trying to figure out how to use a `Function<Out(In...)>` type.

Reply:
Figured it out for real using this “one weird trick”.  Patch tomorrow.

Reply:
The "trick" I found was to use `decltype(&LambdaType::operator())` to get a parameter pack for a pointer to the lambda's `operator()` method, then use that to get the return type and the list of parameters.  It's brilliant in its simplicity.

From there I used another template (by expanding the parameter pack for the list of parameters) to get the first parameter's type.

This prevents the need to declare the element type of the `Vector<>` or the `CFTypeRef` more than once for this template:
```
template<typename MapLambdaType> Vector<typename LambdaTypeTraits<MapLambdaType>::returnType::value_type> makeVector(CFArrayRef, MapLambdaType&&);
```

When used like this in `Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolverCocoa.cpp`:
```
    auto addresses = makeVector(resolvedAddresses, [] (CFDataRef data) -> std::optional<WebCore::IPAddress> {
        return WebCore::IPAddress::fromSockAddrIn6(*reinterpret_cast<const struct sockaddr_in6*>(CFDataGetBytePtr(data)));
    });
```

The only improvement here would be to add support for passing a `WTF::Function<>` instead, but I think this can be done in a later commit (as there is no need for it in this patch).